src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -139,6 +136,42 @@ RobotNavigationObstacleFactory::createObstaclesFromMotionConstraint(
     return obstacles;
 }
 
+std::vector<ObstaclePtr> RobotNavigationObstacleFactory::createGoalObstacles(
+    const Rectangle &goal, bool isFriendly) const
+{
+    // Adjust the goal obstacle radius slightly
+    const double goal_obstacle_radius = ROBOT_MAX_RADIUS_METERS - 0.01;
+
+    std::vector<ObstaclePtr> obstacles;
+
+    // Create top goal post obstacle
+    obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(Stadium(
+        Segment(goal.posXPosYCorner(), goal.negXPosYCorner()), goal_obstacle_radius)));
+
+    // Create bottom goal post obstacle
+    obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(Stadium(
+        Segment(goal.posXNegYCorner(), goal.negXNegYCorner()), goal_obstacle_radius)));
+
+
+    // Create left and right goal wall obstacle
+    if (isFriendly)
+    {
+        Segment segment = Segment(goal.negXPosYCorner(), goal.negXNegYCorner());
+        obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(
+            Stadium(segment, goal_obstacle_radius)));
+    }
+    else
+    {
+        Segment segment = Segment(goal.posXPosYCorner(), goal.posXNegYCorner());
+        obstacles.push_back(std::make_shared<GeomObstacle<Stadium>>(
+            Stadium(segment, goal_obstacle_radius)));
+    }
+
+
```

Comment:
Another minor thing, there's hella whitespace here that could probably be removed.